src/components/Books/Form.jsx

```diff
@@ -1,23 +1,33 @@
 import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
-import { addBook } from '../../redux/books/bookSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { addBook, postBooks } from '../../redux/books/bookSlice';
 
 const Form = () => {
   const dispatch = useDispatch();
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
-  const handleAdd = (event) => {
+  const { bookstore: { isLoading: loading } } = useSelector((store) => store);
+  const handleAdd = async (event) => {
     event.preventDefault();
-    dispatch(addBook({ title, author }));
-    setTitle('');
-    setAuthor('');
+    try {
+      await dispatch(postBooks({ title, author }));
+      dispatch(addBook({ title, author }));
+      setTitle('');
+      setAuthor('');
+    } catch (error) {
+      console.error(error);
```

Comment:
- [ ] for [ javascript best practices ](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) kindly Do not commit console.log to your repo. It’s ugly, it kills performance and it can make confidential data visible to anyone using the browser tools to look at your website. kindly remove it 🙏

Reply:
# doubt 

hi @ichala 

i hope you are doing well, my purpose for having the console.log() for catching errors and also it is conditional 
it will only happen if the application break
so kindly  suggest me some different ways to do that